README.md

```diff
@@ -119,44 +113,83 @@ type OnRetry func(n uint, err error)
 
 Function signature of OnRetry function n = count of tries
 
-#### type RetryOpts
+#### type Option
 
 ```go
-type RetryOpts struct {
-}
+type Option func(*config)
 ```
 
-Struct for configure retry tries - count of tries delay - waiting time units -
-waiting time unit (for tests purpose)
+Option represents an option for retry.
 
-#### func  NewRetryOpts
+#### func  Delay
 
 ```go
-func NewRetryOpts() RetryOpts
+func Delay(delay time.Duration) Option
 ```
-Create new RetryOpts struct with default values default tries are 10 default
-delay are 1e5 default units are microsecond
+Delay set delay between retry default are 1e5 units
 
-#### func (RetryOpts) Delay
+#### func  OnRetryFunction
 
 ```go
-func (opts RetryOpts) Delay(delay time.Duration) RetryOpts
+func OnRetryFunction(onRetryFunction OnRetry) Option
 ```
-Delay setter
+OnRetryFunction function callback are called each retry
 
-#### func (RetryOpts) Tries
+log each retry example:
+
+    retry.Do(
+    	func() error {
+    		return errors.New("some error")
+    	},
+    	retry.OnRetryFunction(func(n unit, err error) {
```

Comment:
I'd name this only `OnRetry`. Similarly, just `RetryIf` instead of `RetryIfFunction`